### PR TITLE
feat(triggers): inotify

### PR DESCRIPTION
**Pull request by l3uddz**
## Summary

inotify trigger can monitor local filesystem folders for new files, renames and deletes.

## Other

* bernard paths are now prefixed with / to be consistent with the configuration examples.
* filters now process excludes first and includes second (it is no longer one or the other).

### CODE REVIEW

File: cmd/autoscan/main.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package main
 import (
 	"errors"
 	"fmt"
+	"github.com/cloudbox/autoscan/triggers/inotify"
```
Comment (m-rots) [inline cmd/autoscan/main.go:6]: move this down to the other triggers

File: triggers/inotify/inotify.go [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+				Err(err).
+				Msg("Failed initialising jobs")
+			return
+		}
+	}
+
+	return trigger, nil
+}
+
+func (d *daemon) StartMonitoring() error {
```
Comment (m-rots) [inline triggers/inotify/inotify.go:89]: should this function be exported?

File: triggers/inotify/inotify.go [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+func (q *queue) process() {
+	// acquire lock
+	q.lock.Lock()
+	defer q.lock.Unlock()
+
+	// sleep if no scans queued
+	if len(q.scans) == 0 {
+		time.Sleep(100 * time.Millisecond)
+		return
+	}
```
Comment (m-rots) [inline triggers/inotify/inotify.go:278]: sleep locks the mutex

**Review by m-rots — CHANGES_REQUESTED** [review]

File: triggers/inotify/inotify.go (hunk 3 above)
Comment (l3uddz) [reply triggers/inotify/inotify.go:278]: thats fine though, as the channel can continue to receive queued "paths", which will be processed on unlock

File: triggers/inotify/inotify.go (hunk 2 above)
Comment (l3uddz) [reply triggers/inotify/inotify.go]: updated this and the equivelant bernard one.

File: cmd/autoscan/main.go (hunk 1 above)
Comment (l3uddz) [reply cmd/autoscan/main.go]: done

**Review by m-rots — DISMISSED** [review]

**Review by m-rots — APPROVED** [review]

**Pull request merged**